state/remoteentities.go

```diff
@@ -118,23 +119,24 @@ func (r *RemoteEntities) ImportRemoteEntity(
 	if token == "" {
 		return errors.NotValidf("empty token for %v in model %v", entity.Id(), sourceModel.Id())
 	}
-	ops := r.importRemoteEntityOps(sourceModel, entity, token)
-	err := r.st.db().RunTransaction(ops)
-	if err == txn.ErrAborted {
-		return errors.AlreadyExistsf(
-			"reference to %s in %s",
-			names.ReadableString(entity),
-			names.ReadableString(sourceModel),
-		)
-	}
-	if err != nil {
-		return errors.Annotatef(
-			err, "recording reference to %s in %s",
-			names.ReadableString(entity),
-			names.ReadableString(sourceModel),
-		)
+	buildTxn := func(int) (ops []txn.Op, _ error) {
+		_, err := r.remoteEntityDoc(sourceModel, entity)
+		if err != nil && err != mgo.ErrNotFound {
+			return nil, errors.Trace(err)
+		}
+		if err == nil {
+			// Token already exists, so remove first.
```

Comment:
If it already exists, can't you just return jujutxn.ErrNoOperations? it appears that you're removing and importing the exact same thing?